client/state/jetpack/credentials/test/reducer.js

```diff
@@ -1,61 +1,123 @@
+			const action = testCredentials( 123000, 'main' );
+			const stateOut = testRequestStatus( stateIn, action );
+
+			expect( stateOut ).toEqual( {
+				123000: {
+					main: true,
+				},
+			} );
+		} );
 
+		test( 'should return false if credentials are valid', () => {
+			const stateIn = {};
+			const action = markCredentialsAsValid( 123000, 'main' );
 			const stateOut = testRequestStatus( stateIn, action );
+
 			expect( stateOut ).toEqual( {
 				123000: {
-					main: 'valid',
+					main: false,
 				},
 			} );
 		} );
 
-		test( 'status should be invalid if the credentials are not okay', () => {
+		test( 'should return false if credentials are invalid', () => {
 			const stateIn = {};
 			const action = markCredentialsAsInvalid( 234000, 'main' );
+			const stateOut = testRequestStatus( stateIn, action );
 
+			expect( stateOut ).toEqual( {
+				234000: {
+					main: false,
+				},
+			} );
+		} );
+
+		test( 'should return true if credentials are valid (or invalid) and we request a new credentials test for same role', () => {
+			const stateIn = {
+				234000: {
+					main: false,
+				},
+			};
+			const action = testCredentials( 234000, 'main' );
 			const stateOut = testRequestStatus( stateIn, action );
+
 			expect( stateOut ).toEqual( {
 				234000: {
-					main: 'invalid',
+					main: true,
+				},
+			} );
+		} );
+
+		test( 'should keep existing roles request statuses after requesting a new credentials test for a new role', () => {
+			const stateIn = {
+				123456: {
+					main: false,
+				},
+			};
+			const action = testCredentials( 123456, 'alternate' );
+			const stateOut = testRequestStatus( stateIn, action );
+
+			expect( stateOut ).toEqual( {
+				123456: {
+					main: false,
+					alternate: true,
 				},
 			} );
 		} );
+	} );
 
-		test( 'status should be pending if we started the credentials test process', () => {
+	describe( 'testStatus', () => {
+		test( 'should return the initial state', () => {
+			expect( testStatus( undefined, {} ) ).toEqual( {} );
+		} );
+
+		test( 'should be valid if the credentials are okay', () => {
 			const stateIn = {};
-			const action = testCredentials( 345000, 'main' );
+			const action = markCredentialsAsValid( 345000, 'main' );
 
-			const stateOut = testRequestStatus( stateIn, action );
+			const stateOut = testStatus( stateIn, action );
 			expect( stateOut ).toEqual( {
 				345000: {
-					main: 'pending',
+					main: 'valid',
 				},
 			} );
 		} );
 
-		test( 'should keep existing roles statuses after requesting a new test', () => {
+		test( 'should be invalid if the credentials are not okay', () => {
+			const stateIn = {};
+			const action = markCredentialsAsInvalid( 234000, 'main' );
+
+			const stateOut = testStatus( stateIn, action );
+			expect( stateOut ).toEqual( {
+				234000: {
+					main: 'invalid',
+				},
+			} );
+		} );
+
+		test( 'should keep previous value if we started a new credentials test for same role', () => {
```

Comment:
Just as above, do we need a test that demonstrates that data is properly stored and maintained for multiple sites at the same time?

Reply:
Thanks! Just added two test case for that :)